extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchBuildAggregator.java

```diff
@@ -55,7 +69,10 @@ public ArrayOfDoublesSketchBuildAggregator(
     this.valueSelectors = valueSelectors.toArray(new BaseDoubleColumnValueSelector[0]);
     values = new double[valueSelectors.size()];
     sketch = new ArrayOfDoublesUpdatableSketchBuilder().setNominalEntries(nominalEntries)
-        .setNumberOfValues(valueSelectors.size()).build();
+                                                       .setNumberOfValues(valueSelectors.size()).build();
+
+    this.canCacheById = this.keySelector.nameLookupPossibleInAdvance();
```

Comment:
I think it might be more correct to use `ColumnCapabilities.areDictionaryValuesUnique` instead of `DimensionDictionarySelector.nameLookupPossibleInAdvance`, because the latter does not necessarily require that dictionary ids uniquely identify lookup values, just that it can be done without the cursor being on the current row that provided the dictionary ids. `IndexedTable` selectors for example use the row number as the dictionary id iirc. The capabilities could be retrieved from the column selector factory available in the factorize methods to pass in I think.